front/ui/ui-charts/src/speedSpaceChart/components/common/DetailsBox.tsx

```diff
@@ -64,36 +69,123 @@ const DetailsBox = ({
+              )
+            )
+          )
+        ),
+      ]
+        .sort((a, b) => a - b)
+        .map((n) => n.toFixed(1))
+    : [];
   return (
     <div
       id="details-box"
-      className={cx('absolute flex flex-col', { block: curveY, hidden: !curveY })}
+      className={cx('absolute flex flex-row', { block: curveY, hidden: !curveY })}
       style={{
         marginTop: boxY,
         marginLeft: boxX,
       }}
     >
-      <span id="base-speed-text">{speedText || '--'}</span>
-      <div>
-        <span id="mareco-speed-text">{ecoSpeedText || '--'}</span>
-        {speedDifferenceText && <span id="speed-difference-text">{speedDifferenceText}</span>}
-      </div>
-      {(energySource || tractionStatus || modeText || effortText) && <hr />}
-      {energySource && <span id="mode-text">{modeText || '--'}</span>}
-      {tractionStatus && <span id="effort-text">{effortText || '--'}</span>}
-      {electricalModeText && (
-        <div id="electrical-mode-text">
-          <span>{electricalModeText || '--'}</span>
-          {electricalProfiles && <span className="ml-2">{electricalProfileText || '--'}</span>}
+      <div id="default-details-box" className="flex flex-col">
+        <span id="base-speed-text">{speedText || '--'}</span>
+        <div>
+          <span id="mareco-speed-text">{ecoSpeedText || '--'}</span>
+          {speedDifferenceText && <span id="speed-difference-text">{speedDifferenceText}</span>}
         </div>
-      )}
-      {powerRestrictions && <span id="power-restriction-text">{powerRestrictionText || '--'}</span>}
-      {declivities && (
-        <>
-          <hr />
-          <span id="previous-gradient-text">{`${previousGradientText} ‰`}</span>
-        </>
-      )}
+        {(energySource || tractionStatus || modeText || effortText) && <hr />}
+        {energySource && <span id="mode-text">{modeText || '--'}</span>}
+        {tractionStatus && <span id="effort-text">{effortText || '--'}</span>}
+        {electricalModeText && (
+          <div id="electrical-mode-text">
+            <span>{electricalModeText || '--'}</span>
+            {electricalProfiles && <span className="ml-2">{electricalProfileText || '--'}</span>}
+          </div>
+        )}
+        {powerRestrictions && (
+          <span id="power-restriction-text">{powerRestrictionText || '--'}</span>
+        )}
+        {declivities && (
+          <>
+            <hr />
+            <span id="previous-gradient-text">{`${previousGradientText} ‰`}</span>
+          </>
+        )}
+      </div>
+      {store.detailsBoxDisplay.etcs &&
+        etcsSpeedValues &&
+        activeEtcsBrakingTypes.length > 0 &&
+        activeEtcsBrakingCurveTypes.length > 0 && (
+          <div id="etcs-details-box" style={{ minHeight: detailsBoxHeight }}>
+            <div
+              id="etcs-grid"
+              style={{ gridTemplateColumns: `repeat(${etcsEndOfCurves.length + 2}, auto)` }}
+            >
```

Comment:
```suggestion
          <div className="etcs-details-box" style={{ minHeight: detailsBoxHeight }}>
            <div
              className="etcs-grid"
              style={{ gridTemplateColumns: `repeat(${etcsEndOfCurves.length + 2}, auto)` }}
            >
```
Here, `id` are only used for CSS; we could use `className` directly.

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/14299/commits/22dc55ad1a33a3f8c72ebe0c5f0a79d38021e3f8.